crates/shared/src/trade_finding/mod.rs

```diff
@@ -36,6 +36,7 @@ pub struct Quote {
     pub out_amount: U256,
     pub gas_estimate: u64,
     pub solver: H160,
+    pub interactions: Vec<InteractionData>,
```

Comment:
Fine to merge as is. But the follow up PR could already store the extra metadata we were talking about (`pre-interactions` and `jit_orders`).
To not clutter the struct with more fields it would then be nice to introduce a separate struct to hold all the information. (maybe `Solution` or `Execution`)

Reply:
Moved to separate struct, similar changes applied to `Estimate` struct which also is used to transfer interactions from quote response.